qa-tools/deploy-gke.sh

```diff
@@ -0,0 +1,127 @@
+usage() {
+  cat << 'EOF'
+  Usage:
+  deploy-gke.sh [SERVICE_ACCOUNT_NAME] [KEY_FILE] [PROJ_ID] (optional)
+  Requirements:
+  * The GKE user/service account user needs the following roles:
+    container.admin compute.admin iam.serviceAccountUser
+    Roles can be granted to a service account via the following :
+    `gcloud projects add-iam-policy-binding <proj-id> --member \ 
+     serviceAccount:sa@proj-id.iam.gserviceaccount.com --role \ 
+     roles/container.admin --role roles/compute.admin --role roles/iam.serviceAccountUser`
+    etc. *
+EOF
+}
+
+if [ $# -lt 2 ]
+then
+	usage
+	exit 0
+fi
+
+
+cleanup() {
+  for container in "${CONTAINERS_TMP[@]}"; do
+    if docker container ls -a --format '{{.Names}}' | grep -Eq "^$container\$"; then
+      docker container rm --force $container
+    fi
+  done
+  for path in "${PATHS_TMP[@]}"; do
+    # Only cleanup tmp paths in /tmp/
+    if [[ -d "${path}" ]] && [[ ${path} =~ ^/tmp/ ]]; then
+      rm -rf "${path}"
+    fi
+  done
+  for file in "${FILES_TMP[@]}"; do
+    if [[ -f "${file}" ]]; then
+      rm -f "${file}"
+    fi
+  done
+}
+
+CONTAINERS_TMP=(gke-deploy)
+PATHS_TMP=()
+FILES_TMP=()
+trap cleanup EXIT
+
+set -o errexit
+postfix=$(cat /dev/urandom | env LC_CTYPE=C tr -dc 'a-z0-9' | fold -w 16 | head -n 1)
+export CLUSTER_NAME=${CLUSTER_NAME-cap-${postfix}}
+export CLUSTER_ZONE=${CLUSTER_ZONE-us-west1-a}
+
+#  We are setting up a zonal cluster, not a regional one  
+
+export NODE_COUNT=3
+export SA_USER=$1
+export KEY_FILE=$2
+export PROJECT=${3-suse-css-platform}
+# Log out from any existing context and log in with the GKE service account
+
+gcloud auth revoke
+gcloud auth activate-service-account $SA_USER --key-file $KEY_FILE --project=$PROJECT
+
+# Clusters on k8s 1.10 and above will no longer get compute-rw and storage-ro scopes
+gcloud config set container/new_scopes_behavior true
+
+# Create GKE cluster 
+gcloud container clusters create ${CLUSTER_NAME} --image-type=UBUNTU --machine-type=n1-standard-4 --zone \
+${CLUSTER_ZONE} --num-nodes=$NODE_COUNT --no-enable-basic-auth --no-issue-client-certificate \
+--no-enable-autoupgrade --metadata disable-legacy-endpoints=true 
```

Comment:
It would be nice to set labels for the owner, to make it easier to track. Something like:

```bash
--labels=owner=$(gcloud config get-value account | tr [:upper:] [:lower:] | tr -c a-z0-9_- _ )
```

If this is for CI, it might be easier to hard code the owner (but for manually-run QA instances that wouldn't work as well).